src/response.js

```diff
@@ -23,7 +23,12 @@ export default class Response {
 		this.url = opts.url;
 		this.status = opts.status || 200;
 		this.statusText = opts.statusText || STATUS_CODES[this.status];
-		this.headers = new Headers(opts.headers);
+
+		if (opts.headers instanceof Headers) {
+			this.headers = opts.headers;
+		} else {
+			this.headers = new Headers(opts.headers);
+		}
```

Comment:
This goes against the Fetch Standard which explicitly says to make a copy of the headers ([<code>Response(_body_, _init_)</code> constructor](https://fetch.spec.whatwg.org/#dom-response), step 6). Instead, you should add a special case in the `Headers` constructor. It should provide a performance boost as well.